packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/backup-offer.class.js

```diff
@@ -0,0 +1,52 @@
+import get from 'lodash/get';
+import { BACKUP_OFFER_PREMIUM } from './backup.constants';
+
+export default class BackupOffer {
+  constructor(offer) {
```

Comment:
Maybe offer could be destructured ? That way we can easily know what property are expected and also we only assign the properties the class needs

Reply:
`offer` is the object returned by the API and all the properties are required. Could you please elaborate on why de-structuring is required?

Reply:
A long time ago, we used to declare class like this, but in some cases it fails working with the datagrid because the Object was initialized with data containing a `$promise` property.
 
So we updated our code to declare only properties needed like this:
```suggestion
  constructor({propertyA, propertyB, propertyC}) {
    Object.assign(this, {propertyA, propertyB, propertyC});
  }
```
Thanks,

Reply:
Makes sense. Thanks for the explanation @cbourgois

Reply:
This is done :)